packages/rx_bloc_cli/mason_templates/bricks/rx_bloc_base/__brick__/bin/server/controllers/authentication_controller.dart

```diff
@@ -0,0 +1,94 @@
+import 'package:shelf/shelf.dart';
+
+import '../models/auth_token.dart';
+import '../repositories/auth_token_repository.dart';
+import '../utils/api_controller.dart';
+import '../utils/server_exceptions.dart';
+
+// ignore_for_file: cascade_invocations
+
+class AuthenticationController extends ApiController {
+  static const _authHeader = 'Authorization';
+  final _authTokenRepository = AuthTokenRepository();
+
+  @override
+  void registerRequests(WrappedRouter router) {
+    router.addRequest(
+      RequestType.POST,
+      '/api/authenticate',
+      _authenticationHandler,
+    );
+    router.addRequest(
+      RequestType.POST,
+      '/api/logout',
+      _logoutHandler,
+    );
+  }
+
+  bool isAuthenticated(Request request) {
```

Comment:
Some endpoints may provide different responses to users that are or aren't authenticated.

One of such would be a home page that displays a login form if they aren't authenticated. Otherwise, it would display a message or something else indicating that he has logged in/authenticated.

For that, instead of letting the developer re-engineer it with every new project, we provide it as a convenience method. If the user does not use the value, it can be ignored. However, if such case exists where the value is of significance, he can access it immediatelly.

Reply:
I understand. Thanks for explanation!